src/System.Private.Interop/src/Shared/__ComObject.cs

```diff
@@ -2234,7 +2234,7 @@ private unsafe object GetDynamicAdapterUsingVariance(RuntimeTypeHandle requested
         /// </summary>
         private unsafe object GetDynamicAdapterInternal(RuntimeTypeHandle requestedType, RuntimeTypeHandle targetType)
         {
-            Debug.Assert(requestedType.HasDynamicAdapterClass());
+            Debug.Assert(requestedType.HasDynamicAdapterClass() || requestedType.IsGenericType());
```

Comment:
Why are you changing System.Private.Interop as part of a WebAssembly change? I don't see why this should be changing.

Reply:
I didn't, just rebasing showing a bunch of changes I didn't make on GitHub. I'll make another attempt to make the history weirdness go away or I'll use a new PR to merge.